python/paddle/fluid/dygraph/dygraph_to_static/ast_transformer.py

```diff
@@ -154,14 +160,26 @@ class BasicApiTransformer(gast.NodeTransformer):
     Class to transform basic API from dygraph to static graph.
     """
 
-    def __init__(self, wrapper_root):
+    def __init__(self, wrapper_root, static_analysis_visitor):
         assert isinstance(
             wrapper_root, AstNodeWrapper
         ), "Input non-AstNodeWrapper node for the initialization of BasicApiTransformer."
+
         self.wrapper_root = wrapper_root
         self.root = wrapper_root.node
+        self.static_analysis_visitor = static_analysis_visitor
+        self.node_to_wrapper_map = self.static_analysis_visitor.get_node_to_wrapper_map(
+        )
         self.class_node_dict = {}
         self.feed_name_to_arg_id = {}
+        self._run_static_visitor()
+        self.name_to_tensor_shape = {}
+
+    def _run_static_visitor(self):
+        # TODO: If Tensor.shape used in sub function
+        var_env = self.static_analysis_visitor.get_var_env()
+        var_env.cur_scope = var_env.cur_scope.sub_scopes[0]
```

Comment:
Thanks.

> Why did you replace var_env.cur_scope with var_env.cur_scope.sub_scopes[0]. It will modify the global var_env.cur_scope, Do you need copy.copy to modify it ?

Thanks! it's a bug and I will modify it according to your suggestion.

> sub_scopes[0] may raise index out of bound because it can be empty.

Usually  there must be 1 sub scope for the  decorated function. But as you said, it's better to consider ` index out of bound `